app/controllers/sites/site_contacts_controller.rb

```diff
@@ -0,0 +1,77 @@
+class Sites::SiteContactsController < ApplicationController
+  before_action :set_site
+  before_action :set_site_contact, only: %i[ show edit update destroy ]
+
+  # GET /site_contacts or /site_contacts.json
+  def index
+    @site_contacts = @site.site_contacts
+    @site_contact = SiteContact.new
+    authorize @site_contacts
+  end
+
+  # GET /site_contacts/1 or /site_contacts/1.json
+  def show
+  end
+
+  # GET /site_contacts/new
+  def new
+    @site_contact = SiteContact.new
+    authorize @site_contact
+  end
+
+  # GET /site_contacts/1/edit
+  def edit
+  end
+
+  # POST /site_contacts or /site_contacts.json
+  def create
+    @site_contact = SiteContact.new(site_contact_params)
+    @site_contact.site_id = @site.id
+    authorize @site_contact
+    if @site_contact.save
+      @site_contact = SiteContact.new
+      flash.now[:notice] = "Contact list is updated"
+    end
+    @site_contacts = @site.site_contacts
+  end
+
+  # PATCH/PUT /site_contacts/1 or /site_contacts/1.json
+  def update
+    if @site_contact.update(site_contact_params)
+      @site_contact = SiteContact.new
+      @site_contacts = @site.site_contacts
+      redirect_to site_site_contacts_path(@site), notice: "Contact was successfully updated." 
```

Comment:
There are models Site and SiteContacts. Routes are created by this:
```
resources :sites do
   resources :site_contacts, module: :sites
end
```
This is why there are two 'site' words in the route.
To remove one site from routes I can rename the model (Contact instead of SiteContract) or write all routes manually in the routes file instead of using the resources statement.
What would you do in this case?

Reply:
If you rename the model to Contact, it does read better. I don't remember, but I'm pretty sure nested resources are expected to work like this.

Reply:
That being said, I don't think it matters a whole lot right now and the routes can be adjusted later. So, I would leave it up to your judgement on how to proceed.